### DISCUSSION TITLE
Using littlefs2 by default in build

### DISCUSSION THREAD

**Discussion by tstenvold** [micropython/micropython#15470] [Using MicroPython] [answered]
I've been looking but I can't quite figure out where I can set the default filesystem format for the PYBv11 to be littlefs2 instead of FAT. 

I want it so that the factory reset also formats the flash sector to littlefs2. 

The only thing I've found so far is to manually create the filesystem, so any help in the right direction would be appreciated.

**Comment by peterhinch** [comment]
I thought the Flash already used littlefs2 - see [this line](https://github.com/micropython/micropython/blob/b4b4d161c2fba54de7f4b297af66a2a807d62f98/ports/stm32/boards/PYBV11/mpconfigboard.mk#L16).

**Reply by tstenvold** [reply]
That's what I was also thinking but a factory reset allows the device to be mounted on a normal computer without any issue, but as soon as I create the filesystem using.

```
import os, vfs, pyb
vfs.umount('/flash')
vfs.VfsLfs2.mkfs(pyb.Flash(start=0))
vfs.mount(pyb.Flash(start=0), '/flash')
os.chdir('/flash')
```

The device is no longer mountable (without Fuse driver), so it's no longer FAT formatted, or am I misunderstanding the problem. It appears that flag only allows the Pyboard to use LFS2. Maybe I need to specifically set `MICROPY_VFS_FAT = 0`

The reason I'm looking into this is, I'm having trouble with resets happening during write operations, corrupting all the files on the filesystem and not just the file I was writing to.

**Reply by peterhinch** [reply]
You're right - I was forgetting about MSC mode as I always disable it in boot.py to avoid file corruption. It must be FAT. I'm not sure how you override this. Bear in mind the SD card, if you're using one.

**Reply by tstenvold** [reply, marked as answer]
After reading through the factoryreset.c it appears that only FAT is supported for a factory reset and if you disable FAT using `MICROPY_VFS_FAT = 0`, one needs to create the filesystem themselves...

So either I'll need to come up with a different method to accomplish this, or I need to implement the factoryreset.c to also support LFS2.

**Comment by tstenvold** [comment]
Additionally, to achieve this, you can do something like 

```
static const char fresh_boot_py[] =
    "import os, vfs, pyb\r\n"
    "vfs.umount('/flash')\r\n"
    "p1 = pyb.Flash(start=0, len=96*1024)\r\n"
    "p2 = pyb.Flash(start=96*1024)\r\n"
    "vfs.VfsFat.mkfs(p1, 'Ripples')\r\n"
    "vfs.VfsLfs2.mkfs(p2)\r\n"
    "vfs.mount(p1, '/flash')\r\n"
    "os.chdir('/flash')\r\n"
    "with open('boot.py', 'w') as f:\r\n"
    "    f.write('import pyb, vfs\\r\\n')\r\n"
    "    f.write('vfs.mount(pyb.Flash(start=96*1024), \\'/data\\')\\r\\n')\r\n"
    "pyb.hard_reset()\r\n";
```

So the factory reset does the hybrid partitioning and then creates a new boot.py that will mount the lfs2 partition after reset.
